Img + run

Krym4s opened the pull request.

Exactlywb commented on tools/simple/main.c:
```diff
@@ -81,6 +82,16 @@ int main (int argc, char **argv)
+  
+  if (img_mode) {
+    input = readBmp(&head1, input);
+    old_filesz = file_size - sizeof(head1);
```
This is not old file size. This should be smth like `file_size_with_no_header`.

Exactlywb commented on include/grasshopper/grasshopper.h:
```diff
@@ -17,6 +17,7 @@ typedef struct Key
 } Key;
 
 enum cipher_mode {
+  NOMODE,
```
How is it possible? Add comment or review this one.

Exactlywb commented on tools/simple/main.c:
```diff
@@ -23,17 +24,57 @@ bool import_key(Key* key, const char* key_file) {
   return true;
 }
 
+enum cipher_mode get_encryption_mode(const char* mode) {
```
Maybe move it to another file like `cmd_arguments.h`

Exactlywb commented:
And remove previous commits (due to 'modes from command line')

Krym4s commented on tools/simple/main.c:
```diff
@@ -100,28 +111,53 @@ int main (int argc, char **argv)
-
   fclose (input);
+
+  if (img_mode) {
```
It is like a text mode, it is not insde a special function. IDK.

Krym4s committed:
Add image encryption

This patch introduces new option for image encryption.  At the moment .bmp
image format is supported.  BMP header reads and is not going to be encrypted.

Some notes are fixed too.
